exonum-java-binding/core/rust/integration_tests/src/vm.rs

```diff
@@ -137,27 +158,30 @@ pub fn get_libpath() -> String {
 }
 
 pub fn get_fake_service_artifact_path() -> String {
-    project_root_dir()
+    java_binding_parent_root_dir()
         .join("fake-service/target/fake-service-artifact.jar")
         .to_str()
         .expect(CONVERSION_FAILED_MESSAGE)
         .to_owned()
 }
 
-fn rust_project_root_dir() -> PathBuf {
-    Path::new(env!("CARGO_MANIFEST_DIR"))
-        .join("..")
-        .canonicalize()
-        .unwrap()
-}
-
-fn project_root_dir() -> PathBuf {
+fn java_binding_parent_root_dir() -> PathBuf {
     rust_project_root_dir()
         .join("../..")
         .canonicalize()
         .unwrap()
 }
 
+/// The path to the root directory of the Rust parent module.
+fn rust_project_root_dir() -> PathBuf {
+    project_root_dir().join("..").canonicalize().unwrap()
+}
+
+/// The path to `integration_tests` root directory.
+fn project_root_dir() -> &'static Path {
```

Comment:
Does it have to return a reference? If so when is the object destroyed?

Reply:
I guess we need to return `PathBuf`, not `&Path`.